pilot/pkg/config/kube/crdclient/gen/types.go.tmpl

```diff
@@ -152,17 +157,29 @@ func delete(ic versionedclient.Interface, sc gatewayapiclient.Interface, typ con
 	}
 }
 
-var translationMap = map[config.GroupVersionKind]func(r runtime.Object) config.Config{
+var translationMap = map[config.GroupVersionKind]func(r runtime.Object) config.Config {
 {{- range . }}
 	collections.{{ .VariableName }}.Resource().GroupVersionKind(): func(r runtime.Object) config.Config {
 		obj := r.(*{{ .ClientImport }}.{{ .Kind }})
+		{{- if .WithFieldPathMap }}
+		annots := obj.Annotations 
+		fieldPathMap, err := buildFieldPathMap(obj)
+		if err != nil {
+			scope.Errorf("failed to build field path map for object %s/%s: %v", obj.Namespace, obj.Name, err)
+		} else {
+			if annots == nil {
+				annots = map[string]string{}
+			}
+			annots[file.FieldMapKey] = fieldPathMap 
```

Comment:
I don't see why this "field map" belongs on an annotation

Reply:
it is the same trick that when the istio analyzer analyses a local resource file, it constructs the "field map" for the resource and stores it in the annotation(https://github.com/istio/istio/blob/284def0c93cc2dc37b07a1d7acd74f26eb24fb63/pilot/pkg/config/file/store.go#L433)